src/Counter.php

```diff
@@ -40,14 +42,16 @@ final class Counter implements CounterInterface
      * @param int $storageTtlInSeconds Storage TTL. Should be higher than `$periodInSeconds`.
      * @param string $storagePrefix Storage prefix.
      * @param TimerInterface|null $timer Timer instance to get current time from.
+     * @param int $maxCASAttempts Maximum number of times to retry saveIfNotExists/saveCompareAndSwap operations before returning an error.
      */
     public function __construct(
         private StorageInterface $storage,
         private int $limit,
         private int $periodInSeconds,
         private int $storageTtlInSeconds = self::DEFAULT_TTL,
         private string $storagePrefix = self::ID_PREFIX,
-        TimerInterface|null $timer = null
+        TimerInterface|null $timer = null,
+        private int $maxCASAttempts = self::DEFAULT_MAX_CAS_ATTEMPTS
```

Comment:
```suggestion
        private int $maxCasAttempts = self::DEFAULT_MAX_CAS_ATTEMPTS
```

Use camelCase notation, including abbreviations (see [docs](https://github.com/yiisoft/docs/blob/master/010-code-style.md#names))